website/source/docs/internals/discovery-chain.html.md

```diff
@@ -0,0 +1,217 @@
+[targets](#discoverytarget). Nodes are the compiled representation of various
+discovery chain stages and targets are instructions on how to use the [health
+API](/api/health.html#list-nodes-for-connect-capable-service) to retrieve
+relevant service instance lists.
+
+You should traverse the nodes starting with [`StartNode`](#startnode). The
+nodes can be resolved by name using the [`Nodes`](#nodes) field. Targets can be
+resolved by name using the [`Targets`](#targets) field.
+
+- `ServiceName` `(string)` - The requested service.
+- `Namespace` `(string)` - The requested namespace.
+- `Datacenter` `(string)` - The requested datacenter.
+
+- `CustomizationHash` `(string: <optional>)` - A unique hash of any overrides
+  that affected the compilation of the discovery chain.
+
+    If set, this value should be used to prefix/suffix any generated load
+    balancer data plane objects to avoid sharing customized and non-customized
+    versions.
+
+- `Protocol` `(string)` - The overall protocol shared by everything in the
+  chain.
+
+- `StartNode` `(string)` - The first key into the `Nodes` map that should be
+  followed when traversing the discovery chain.
+
+- `Nodes` `(map<string|DiscoveryGraphNode>)` - All nodes available for traversal in
+  the chain keyed by a unique name. You can walk this by starting with
+  `StartNode`.
+
+    -> The names should be treated as opaque values and are only guaranteed to be
+    consistent within a single compilation.
+
+- `Targets` `(map<string|DiscoveryTarget>)` - A list of all targets used in this chain.
+
+    -> The names should be treated as opaque values and are only guaranteed to be
+    consistent within a single compilation.
+
+#### `DiscoveryGraphNode`
+
+A single node in the compiled discovery chain.
+
+- `Type` `(string)` - The type of the node. Valid values are: `router`,
+  `splitter`, and `resolver`.
+
+- `Name` `(string)` - The unique name of the node.
+
+- `Routes` `(array<DiscoveryRoute>)` - Only set for `Type:router`. List of routes to
+  render.
+
+  - `Definition` `(ServiceRoute)` - Relevant portion of underlying
+    `service-router`
+    [route](/docs/agent/config-entries/service-router.html#routes).
+
+  - `NextNode` `(string)` - The name of the next node in the chain in [`Nodes`](#nodes).
+
+- `Splits` `(array<DiscoverySplit>)` - Only set for `Type:splitter`. List of traffic
+  splits.
+
+  - `Weight` `(float32)` - Copy of underlying `service-splitter`
+    [`weight`](/docs/agent/config-entries/service-splitter.html#weight) field.
+
+  - `NextNode` `(string)` - The name of the next node in the chain in [`Nodes`](#nodes).
+
+- `Resolver` `(DiscoveryResolver: <optional>)` - Only set for `Type:resolver`. How
+  to resolve the service instances.
+
+  - `Default` `(bool)` - Set to true if no `service-resolver` config entry is
+    defined for this node and the default was synthesized.
+
+  - `ConnectTimeout` `(duration)` - Copy of the underlying `service-resolver`
```

Comment:
It's actually an integer in nanoseconds it seems. I'd expect `(duration)` to be a string like `5s`